opencensus/trace/exporters/jaeger_exporter.py

```diff
@@ -0,0 +1,451 @@
+        self.username = username
+        self.password = password
+        self.agent_client = self._agent_client
+        self.collector = self._collector
+
+    @property
+    def _agent_client(self):
+        return AgentClientUDP(
+            host_name=self.agent_host_name,
+            port=self.agent_port)
+
+    @property
+    def _collector(self):
+        if self.host_name is None or self.port is None:
+            return None
+
+        thrift_url = 'http://{}:{}{}{}'.format(
+            self.host_name,
+            self.port,
+            self.endpoint,
+            DEFAULT_ENDPOINT)
+
+        auth = None
+        if self.username != '' and self.password != '':
+            auth = (self.username, self.password)
+
+        return Collector(thrift_url=thrift_url, auth=auth)
+
+    def emit(self, span_datas):
+        """
+        :type span_datas: list of :class:
+            `~opencensus.trace.span_data.SpanData`
+        :param list of opencensus.trace.span_data.SpanData span_datas:
+            SpanData tuples to emit
+        """
+        trace = span_data.format_legacy_trace_json(span_datas)
+
+        jaeger_spans = self.translate_to_jaeger(trace)
+
+        batch = jaeger.Batch(
+            spans=jaeger_spans,
+            process=jaeger.Process(serviceName=self.service_name))
+
+        if self.collector is not None:
+            self.collector.export(batch)
+        self.agent_client.export(batch)
+
+    def export(self, span_datas):
+        """Export the trace. Send trace to transport, and transport will call
+        exporter.emit() to actually send the trace to the specified tracing
+        backend.
+
+        :type span_datas: list of :class:
+            `~opencensus.trace.span_data.SpanData`
+        :param list of opencensus.trace.span_data.SpanData span_datas:
+            SpanData tuples to export
+        """
+        self.transport.export(span_datas)
+
+    def translate_to_jaeger(self, trace):
+        """Translate the spans json to Jaeger format.
+
+        :type trace: dict
+        :param trace: Trace dictionary
+
+        :rtype: list
+        :returns: List of jaeger.Span objects generated by Thrift.
+        """
+
+        trace_id = trace.get('traceId', '')
+        spans = trace.get('spans')
+        jaeger_spans = []
+
+        for span in spans:
+            start_datetime = datetime.datetime.strptime(
+                span.get('startTime'), ISO_DATETIME_REGEX)
+            start_microsec = calendar.timegm(start_datetime.timetuple()) * 1000
+
+            end_datetime = datetime.datetime.strptime(
+                span.get('endTime'), ISO_DATETIME_REGEX)
+            end_microsec = calendar.timegm(end_datetime.timetuple()) * 1000
+
+            duration_microsec = end_microsec - start_microsec
+
+            tags = _extract_tags(span)
+
+            status = span.get('status', None)
+            if status is not None:
+                tags.append(jaeger.Tag(
+                    key='status.code',
```

Comment:
Can you use the error tags from the [opentracing spec](https://github.com/opentracing/specification/blob/master/semantic_conventions.md#captured-errors)?
Also, can you add an error=true tag so that error spans will render as errors in the Jaeger UI?

Reply:
I meant for non-OK statuses you should additionally add an `error=True` tag.

I misspoke earlier, the other error information is stored as span logs and not tags. It looks like the only error related tag in OT is `error`. Maybe in a follow up PR (or this one) we should additionally encode the Status message and Stack Trace information into span logs.